src/image.py

```diff
@@ -84,19 +81,42 @@ def build(self):
         if self.info.get("base_image_uri"):
             self.build_args["BASE_IMAGE"] = self.info["base_image_uri"]
 
+        if self.ecr_url:
+            self.build_args["FIRST_STAGE_IMAGE"] = self.ecr_url
+
         if self.info.get("extra_build_args"):
             self.build_args.update(self.info.get("extra_build_args"))
-
+        
         if self.info.get("labels"):
             self.labels.update(self.info.get("labels"))
+        
+        print(f"self.build_args {self.build_args}")
+        print(f"self.labels {self.labels}")
 
-        with open(self.context.context_path, "rb") as context_file:
-            response = []
-
-            for line in self.client.build(
-                fileobj=context_file,
+    def build(self):
+        """
+        The build function builds the specified docker image
+        """
+        self.summary["start_time"] = datetime.now()
+        self.pre_build_configuration()
+        print(f"self.context {self.context}")
+        if self.context:
+            with open(self.context.context_path, "rb") as context_file:
+                print("within context")
+                self.docker_build(fileobj=context_file, custom_context=True)
+                self.context.remove()  
+        else:
+            print("out of context")
+            self.docker_build()
+        #check the size after image is built.
+        self.image_size_check()
```

Comment:
We need to return `self.build_status`. If we do not return that, the OutputFormatter fails as `futures[image].result()` remains None